src/source.cpp

```diff
@@ -355,30 +344,57 @@ void free_memory_source()
   model::external_sources.clear();
 }
 
-// fill the source bank from the external source
-void fill_source_bank_custom_source()
+//Load custom source library
+void load_custom_source_library()
 {
 #ifdef HAS_DYNAMIC_LINKING
+
   // Open the library
-  auto source_library = dlopen(settings::path_source_library.c_str(), RTLD_LAZY);
-  if (!source_library) {
+  model::source_library = dlopen(settings::path_source_library.c_str(), RTLD_LAZY);
+  if (!model::source_library) {
     fatal_error("Couldn't open source library " + settings::path_source_library);
   }
 
   // reset errors
   dlerror();
 
   // get the function from the library
-  using sample_t = Particle::Bank (*)(uint64_t* seed);
-  auto sample_source = reinterpret_cast<sample_t>(dlsym(source_library, "sample_source"));
+  //using sample_t = Particle::Bank (*)(uint64_t* seed);
+  model::sample_source = reinterpret_cast<model::sample_t>(dlsym(model::source_library, "sample_source"));
 
   // check for any dlsym errors
   auto dlsym_error = dlerror();
   if (dlsym_error) {
-    dlclose(source_library);
+    dlclose(model::source_library);
     fatal_error(fmt::format("Couldn't open the sample_source symbol: {}", dlsym_error));
   }
 
+#else
+  fatal_error("Custom source libraries have not yet been implemented for "
+    "non-POSIX systems");
+#endif
+
+}
+
+//Release custom source library
```

Comment:
Generally we put the description of a function with its declaration in the .h file (using a Doxygen comment). I would move the comments for the other functions as well.